api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_stats.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import os
+from concurrent import futures
+from typing import Any, Generator, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import tiledbsoma as soma
+
+from ..util._eager_iter import _EagerIterator
+from ._online import MeanAccumulator, MeanVarianceAccumulator
+
+
+def mean_variance(
+    query: soma.ExperimentAxisQuery,
+    layer: str = "raw",
+    axis: int = 0,
+    calculate_mean: bool = False,
+    calculate_variance: bool = False,
+    ddof: int = 1,
+) -> pd.DataFrame:
+    """
+    Calculate  mean and/or variance along the `obs` axis from query results. Calculations
+    are done in an accumulative chunked fashion.
+
+    Args:
+        query:
+            A SOMA query, specifying the obs/var selection over which mean and variance are calculated.
+
+        layer:
+            X layer used, e.g., `raw`
+
+        axis:
+           Axis or axes along which the statistics are computed
+
+        calculate_mean:
+            If `True it calculates mean, otherwise skips calculation
+
+        calculate_variance:
+            If `True it calculates variance, otherwise skips calculation
+
+        ddof:
+            "Delta Degrees of Freedom": the divisor used in the calculation for variance is N - ddof, where N represents the number of elements.
+
+    Returns:
+        Pandas DataFrame indexed by the obs `soma_joinid` and with columns
+       `mean` (if `calculate_mean = True`), and `variance` (if `calculate_variance = True`)
+
+    Lifecycle:
+        experimental
+    """
+
+    if axis not in (0, 1):
+        raise ValueError("axis must be 0 or 1")
+
+    if calculate_mean is False and calculate_variance is False:
+        raise ValueError("At least one of `calculate_mean` or `calculate_variance` must be True")
+
+    if query.n_obs == 0 or query.n_vars == 0:
+        raise ValueError("The query cannot yield an empty result")
+
+    n_dim_0 = query.n_obs if axis == 1 else query.n_vars
+    n_dim_1 = query.n_vars if axis == 1 else query.n_obs
+
+    n_batches = 1
+    n_samples = np.array([n_dim_1], dtype=np.int64)
+
+    idx = pd.Index(data=query.obs_joinids() if axis == 1 else query.var_joinids(), name="soma_joinid")
+
+    def iterate() -> Generator[Tuple[npt.NDArray[np.int64], Any], None, None]:
+        max_workers = (os.cpu_count() or 4) + 2
+        with futures.ThreadPoolExecutor(max_workers=max_workers) as pool:
```

Comment:
small simplification (nit, optional):  EagerIterator can use at most a single thread.  So you could remove the code that computes max_workers (line 73), and just hard-wire a `max_workers=1` into line 74.